src/Component/EventDispatcher/src/DependencyInjection/Compiler/RegisterListenerPass.js

```diff
@@ -0,0 +1,102 @@
+const CompilerPassInterface = Jymfony.DependencyInjection.CompilerPassInterface;
+const EventSubscriberInterface = Jymfony.EventDispatcher.EventSubscriberInterface;
+const EventDispatcher = Jymfony.EventDispatcher.EventDispatcher;
+const ServiceClosureArgument = Jymfony.DependencyInjection.ServiceClousureArgument;
+
+class RegisterListenerPass extends implementationOf(CompilerPassInterface) {
+    __constructor (dispatcherService = 'event_dispatcher', listenerTag = 'kernel.event_listener', subscriberTag = 'kernel.event_subscriber') {
+        this.dispatcherService = dispatcherService;
+        this.listenerTag = listenerTag;
+        this.subscriberTag = subscriberTag;
+    }
+
+    /**
+     *
+     * @param {Jymfony.Kernel.ContainerBuilder} container
+     */
+    process(container) {
+        if (! container.hasDefinition(this.dispatcherService) && !container.hasAlias(this.dispatcherService)) {
+            return;
+        }
+
+        let definition = container.findDefinition(this.dispatcherService);
+
+        for (let [id, events] of __jymfony.getEntries(container.findTaggedServiceIds(this.listenerTag))) {
+            for (let event of events) {
+                let priority = event['priority'] !== undefined ? event['priority'] : 0;
```

Comment:
Prefer dot notation (`event.priority` instead of `event['priority']`) if using non-numeric, non-dynamic properties.